### ISSUE TITLE
Cannot install Rook on K8s using kubeadm-dind v1.5- mons stuck in pending state

### ISSUE THREAD

**Issue by dangula** [rook/rook#825]
## Repro steps
1.  Start k8s 1.5 via kubeadm-dind
* run `export KUBE_VERSION=v1.5`
* run `tests/script/kubeadm-dind.sh up`
2. using yaml files under  `demo/kubernetes/1.5` install rook 
* `kubectl create -f rook-operator.yaml `
* `kubectl create -f rook-cluster.yaml`
#### Additional Info
```
NAMESPACE     NAME                                    READY     STATUS    RESTARTS   AGE
default       rook-operator-2406728969-bhpjn          1/1       Running   0          3m
kube-system   dummy-2088944543-d81zr                  1/1       Running   1          23m
kube-system   etcd-kube-master                        1/1       Running   1          23m
kube-system   kube-apiserver-kube-master              1/1       Running   0          23m
kube-system   kube-controller-manager-kube-master     1/1       Running   0          22m
kube-system   kube-discovery-2220406887-5dqsh         1/1       Running   1          23m
kube-system   kube-dns-1873809965-tbmnr               3/3       Running   0          21m
kube-system   kube-proxy-5k1mw                        1/1       Running   0          22m
kube-system   kube-proxy-b5p4n                        1/1       Running   0          22m
kube-system   kube-proxy-j68sx                        1/1       Running   0          22m
kube-system   kube-scheduler-kube-master              1/1       Running   0          22m
kube-system   kubernetes-dashboard-3203962772-05zf7   1/1       Running   0          21m
rook          rook-ceph-mon0-38snc                    0/1       Pending   0          3m

```

##### snippet from rook-operator log
```
2017-07-19 07:54:08.804350 I | op-mon: creating mon secrets for a new cluster
2017-07-19 07:54:08.804755 I | exec: Running command: ceph-authtool --create-keyring /var/lib/rook/rook/mon.keyring --gen-key -n mon. --cap mon 'allow *'
2017-07-19 07:54:08.958949 I | exec: Running command: ceph-authtool --create-keyring /var/lib/rook/rook/client.admin.keyring --gen-key -n client.admin --set-uid=0 --cap mon 'allow *' --cap osd 'allow *' --cap mgr 'allow *' --cap mds 'allow'
2017-07-19 07:54:08.988551 I | op-mon: saved rook-admin secret
2017-07-19 07:54:08.992866 I | op-mon: saved mon endpoints to config map map[endpoints: maxMonId:-1]
2017-07-19 07:54:08.993217 I | cephmon: writing config file /var/lib/rook/rook/rook.config
2017-07-19 07:54:08.993347 I | cephmon: generated admin config in /var/lib/rook/rook
2017-07-19 07:54:08.993542 I | cephmon: writing config file /var/lib/rook/rook/rook.config
2017-07-19 07:54:08.993663 I | cephmon: generated admin config in /var/lib/rook/rook
2017-07-19 07:54:08.998413 I | op-mon: there are 3 nodes available for mons (existing mons=0)
2017-07-19 07:54:09.000631 I | op-mon: Found 3 running nodes without mons
2017-07-19 07:54:09.010630 I | op-mon: waiting for mon rook-ceph-mon0 to start. status=
2017-07-19 07:54:15.021688 I | op-mon: waiting for mon rook-ceph-mon0 to start. status=Pending
2017-07-19 07:54:21.033115 I | op-mon: waiting for mon rook-ceph-mon0 to start. status=Pending
2017-07-19 07:54:27.044168 I | op-mon: waiting for mon rook-ceph-mon0 to start. status=Pending
2017-07-19 07:54:33.058842 I | op-mon: waiting for mon rook-ceph-mon0 to start. status=Pending
2017-07-19 07:54:39.071178 I | op-mon: waiting for mon rook-ceph-mon0 to start. status=Pending
2017-07-19 07:54:45.084031 I | op-mon: waiting for mon rook-ceph-mon0 to start. status=Pending
2017-07-19 07:54:51.097377 I | op-mon: waiting for mon rook-ceph-mon0 to start. status=Pending
2017-07-19 07:54:57.109652 I | op-mon: waiting for mon rook-ceph-mon0 to start. status=Pending
2017-07-19 07:55:03.121003 I | op-mon: waiting for mon rook-ceph-mon0 to start. status=Pending
2017-07-19 07:55:09.136632 I | op-mon: waiting for mon rook-ceph-mon0 to start. status=Pending
2017-07-19 07:55:15.160518 I | op-mon: waiting for mon rook-ceph-mon0 to start. status=Pending
2017-07-19 07:55:21.172106 I | op-mon: waiting for mon rook-ceph-mon0 to start. status=Pending
2017-07-19 07:55:27.185235 I | op-mon: waiting for mon rook-ceph-mon0 to start. status=Pending
2017-07-19 07:55:33.198358 I | op-mon: waiting for mon rook-ceph-mon0 to start. status=Pending
2017-07-19 07:55:39.209228 E | operator: failed to create cluster rook in namespace rook. failed to start the mons. failed to start mon pods. failed to start pod rook-ceph-mon0. timed out waiting for pod rook-ceph-mon0 to start 
```

##### output from kubectl describe pod rook-ceph-mon0
```
ame:		rook-ceph-mon0-38snc
Namespace:	rook
Node:		/
Labels:		app=rook-ceph-mon
		mon=rook-ceph-mon0
		mon_cluster=rook
Annotations:	kubernetes.io/created-by={"kind":"SerializedReference","apiVersion":"v1","reference":{"kind":"ReplicaSet","namespace":"rook","name":"rook-ceph-mon0","uid":"7215971e-6c57-11e7-8697-02420ac00002","apiVe...
		rook_version=master
Status:		Pending
IP:		
Controllers:	ReplicaSet/rook-ceph-mon0
Containers:
  rook-ceph-mon:
    Image:	rook/rook:master
    Port:	6790/TCP
    Command:
      /bin/sh
      -c
      sleep 5; /usr/local/bin/rook mon --config-dir=/var/lib/rook --name=rook-ceph-mon0 --port=6790 --fsid=1a20d259-8572-4357-a7b3-dc983669e655
    Environment:
      ROOKD_PRIVATE_IPV4:		 (v1:status.podIP)
      ROOKD_CLUSTER_NAME:		rook
      ROOKD_MON_ENDPOINTS:		<set to the key 'endpoints' of config map 'mon-config'>		Optional: false
      ROOKD_MON_SECRET:			<set to the key 'mon-secret' in secret 'rook-ceph-mon'>		Optional: false
      ROOKD_ADMIN_SECRET:		<set to the key 'admin-secret' in secret 'rook-ceph-mon'>	Optional: false
      ROOKD_CEPH_CONFIG_OVERRIDE:	/etc/rook/override.conf
    Mounts:
      /etc/rook from rook-config-override (rw)
      /var/lib/rook from rook-data (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-pn12s (ro)
Conditions:
  Type		Status
  PodScheduled 	False 
Volumes:
  rook-data:
    Type:	EmptyDir (a temporary directory that shares a pod's lifetime)
    Medium:	
  rook-config-override:
    Type:	ConfigMap (a volume populated by a ConfigMap)
    Name:	rook-config-override
    Optional:	false
  default-token-pn12s:
    Type:	Secret (a volume populated by a Secret)
    SecretName:	default-token-pn12s
    Optional:	false
QoS Class:	BestEffort
Node-Selectors:	kubernetes.io/hostname=kube-master
Tolerations:	<none>
Events:
  FirstSeen	LastSeen	Count	From			SubObjectPath	Type		Reason			Message
  ---------	--------	-----	----			-------------	--------	------			-------
  7m		8s		29	default-scheduler			Warning		FailedScheduling	pod (rook-ceph-mon0-38snc) failed to fit in any node
fit failure summary on nodes : MatchNodeSelector (2), PodToleratesNodeTaints (1)
```

**Comment by jbw976** [comment]
Is this related to #791?

**Comment by dangula** [comment]
Its kind of related, that bug is closed because 1.5 worked with minikube. But it is not working on kubeadm-dind for CI framework. Also one minor difference minikube states k8s 1.5.3 and kubeadm-dind is starting 1.5.4(1.5.4 is not supported for minikube)
